tempesta_fw/pool.c

```diff
@@ -99,6 +126,26 @@ EXPORT_SYMBOL(tfw_pool_realloc);
 void
 tfw_pool_free(TfwPool *p)
 {
-	free_pages((unsigned long)p, p->order);
+	TfwPoolChunk *chunk, *next;
+
+	chunk = p->head;
+	while (chunk != NULL) {
+		next = chunk->next;
+		free_pages((unsigned long)chunk, chunk->order);
+		chunk = next;
+	}
 }
 EXPORT_SYMBOL(tfw_pool_free);
+
+void
+tfw_pool_try_free(TfwPool *p, void *ptr, size_t n)
```

Comment:
Please place the function just after tfw_pool_realloc() - it's good to keep logically near functions together